Attempting to read specific fields with compound values panics for interface conversion

*Discovered in version: 0.5.2*

Simply put, reading a field with value of type map or list causes the app to panic and dump the stack onto console.

```
dexterous@dexters-lab01:~$ vault version
Vault v0.5.2

$ vault read auth/token/lookup-self
Key             Value
accessor     
creation_time   1.458762732e+09
creation_ttl    2.592e+06
display_name    ldap-smhatre
id              afea4ddf-db8c-85d5-8f25-bbaf525f85da
meta            map[policies:develop,manage-secrets username:smhatre]
num_uses        0
orphan          true
path            auth/ldap/login/smhatre
policies        [develop manage-secrets default]
role         
ttl             1.290739e+06

dexterous@dexters-lab01:~$ vault read -field=meta auth/token/lookup-self
panic: interface conversion: interface is map[string]interface {}, not string

goroutine 1 [running]:
panic(0x1218660, 0xc8205f7ec0)
        /goroot/src/runtime/panic.go:464 +0x3e6
github.com/hashicorp/vault/command.(*ReadCommand).Run(0xc8202d4820, 0xc82000a3f0, 0x1, 0x1, 0xc8202c5400)
        /gopath/src/github.com/hashicorp/vault/command/read.go:72 +0xa63
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc8200d2240, 0xc8200d2240, 0x0, 0x0)
        /gopath/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:153 +0x56e
github.com/hashicorp/vault/cli.RunCustom(0xc82000a3d0, 0x3, 0x3, 0xc8202e6030, 0xc8200161d8)
        /gopath/src/github.com/hashicorp/vault/cli/main.go:44 +0x454
github.com/hashicorp/vault/cli.Run(0xc82000a3d0, 0x3, 0x3, 0x1403ad0)
        /gopath/src/github.com/hashicorp/vault/cli/main.go:11 +0x4c
main.main()
        /gopath/src/github.com/hashicorp/vault/main.go:10 +0x60

dexterous@dexters-lab01:~$ vault read -field=policies auth/token/lookup-self
panic: interface conversion: interface is []interface {}, not string

goroutine 1 [running]:
panic(0x1218660, 0xc820309440)
        /goroot/src/runtime/panic.go:464 +0x3e6
github.com/hashicorp/vault/command.(*ReadCommand).Run(0xc82030eaa0, 0xc82000a3f0, 0x1, 0x1, 0xc8203262f0)
        /gopath/src/github.com/hashicorp/vault/command/read.go:72 +0xa63
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc8200f6240, 0xc8200f6240, 0x0, 0x0)
        /gopath/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:153 +0x56e
github.com/hashicorp/vault/cli.RunCustom(0xc82000a3d0, 0x3, 0x3, 0xc820013b90, 0xc8200161d8)
        /gopath/src/github.com/hashicorp/vault/cli/main.go:44 +0x454
github.com/hashicorp/vault/cli.Run(0xc82000a3d0, 0x3, 0x3, 0x1403ad0)
        /gopath/src/github.com/hashicorp/vault/cli/main.go:11 +0x4c
main.main()
        /gopath/src/github.com/hashicorp/vault/main.go:10 +0x60
```

--- a/command/read.go
+++ b/command/read.go
@@ -70,9 +70,9 @@ func (c *ReadCommand) Run(args []string) int {
 			// directly print the message. If mitchellh/cli exposes method
 			// to print without CR, this check needs to be removed.
 			if reflect.TypeOf(c.Ui).String() == "*cli.BasicUi" {
-				fmt.Fprintf(os.Stdout, val.(string))
+				fmt.Fprintf(os.Stdout, fmt.Sprintf("%v", val))
 			} else {
-				c.Ui.Output(val.(string))
+				c.Ui.Output(fmt.Sprintf("%v", val))
 			}
 			return 0
 		} else {

--- a/command/write.go
+++ b/command/write.go
@@ -82,9 +82,9 @@ func (c *WriteCommand) Run(args []string) int {
 			// directly print the message. If mitchellh/cli exposes method
 			// to print without CR, this check needs to be removed.
 			if reflect.TypeOf(c.Ui).String() == "*cli.BasicUi" {
-				fmt.Fprintf(os.Stdout, val.(string))
+				fmt.Fprintf(os.Stdout, fmt.Sprintf("%v", val))
 			} else {
-				c.Ui.Output(val.(string))
+				c.Ui.Output(fmt.Sprintf("%v", val))
 			}
 			return 0
 		} else {

